pgcli/pgcompleter.py

```diff
@@ -296,48 +319,62 @@ def get_completions(self, document, complete_event, smart_completion=None):
                         not word_before_cursor.startswith('pg_')):
                     views = [v for v in views if not v.startswith('pg_')]
 
-                views = self.find_matches(word_before_cursor, views)
+                views = self.find_matches(word_before_cursor, views,
+                                          meta='view')
                 completions.extend(views)
 
             elif suggestion['type'] == 'alias':
                 aliases = suggestion['aliases']
-                aliases = self.find_matches(word_before_cursor, aliases)
+                aliases = self.find_matches(word_before_cursor, aliases,
+                                            meta='table alias')
                 completions.extend(aliases)
 
             elif suggestion['type'] == 'database':
-                dbs = self.find_matches(word_before_cursor, self.databases)
+                dbs = self.find_matches(word_before_cursor, self.databases,
+                                        meta='database')
                 completions.extend(dbs)
 
             elif suggestion['type'] == 'keyword':
                 keywords = self.find_matches(word_before_cursor, self.keywords,
                                              start_only=True,
-                                             fuzzy=False)
+                                             fuzzy=False,
+                                             meta='keyword')
                 completions.extend(keywords)
 
             elif suggestion['type'] == 'special':
-                special = self.find_matches(word_before_cursor,
-                                            self.special_commands,
+                if not self.pgspecial:
+                    continue
```

Comment:
PGCompleter now accepts an optional PGSpecial object during initialization. When usin g PGCompleter outside of pgcli, or in the unit tests, the PGSpecial object might not be supplied. In that case, I didn't want to error out when `pgcli.packages.sqlcompletion.suggest_type` suggested a special command, e.g.:

```
In [1]: from pgcli.pgcompleter import PGCompleter

In [2]: from prompt_toolkit.document import Document

In [3]: completer = PGCompleter()

In [4]: completer.get_completions(Document(text=u'\\', cursor_position=1), {})
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-4-8eceb1def582> in <module>()
----> 1 completer.get_completions(Document(text=u'\\', cursor_position=1), {})

c:\users\dg\documents\python\pgcli\pgcli\pgcompleter.pyc in get_completions(self, document, complete_event, smart_completion)
    346                 #    continue
    347
--> 348                 commands = self.pgspecial.commands
    349                 cmd_names = commands.keys()
    350                 desc = [commands[cmd].description for cmd in cmd_names]

AttributeError: 'NoneType' object has no attribute 'commands'
```